File: Jack2/src/main/java/ch/chassaing/jack/token/SymbolType.java
package ch.chassaing.jack.token;

public enum SymbolType
{
    LEFT_BRACE('{'), RIGHT_BRACE('}'),
    LEFT_PAREN('('), RIGHT_PAREN(')'),
    LEFT_BRACKET('['), RIGHT_BRACKET(']'),
    DOT('.'), COMMA(','), SEMICOLON(';'),
    PLUS('+'), MINUS('-'),
    STAR('*'), SLASH('/'),
    AMP('&'), PIPE('|'),
    LT('<'), GT('>'), EQUAL('='),
    NEG('~');

    public final char repr;

    SymbolType(char repr) {this.repr = repr;}
}

File: JackCompiler/src/main/kotlin/ch/chassaing/jack/SymbolType.kt
package ch.chassaing.jack

enum class SymbolType(val c : Char)  {

    LEFT_BRACE('{'), RIGHT_BRACE('}'),
    LEFT_PAREN('('), RIGHT_PAREN(')'),
    LEFT_BRAKET('['), RIGHT_BRAKET(']'),
    DOT('.'), COMMA(','), SEMICOLON(';'),
    PLUS('+'), MINUS('-'),
    STAR('*'), SLASH('/'),
    AMP('&'), PIPE('|'),
    LT('<'), GT('>'), EQUAL('='),
    NEG('~');
}